lib/cosmos/tools/cmd_tlm_server/interface_thread.rb

```diff
@@ -89,14 +104,20 @@ def start
 
     # Disconnect from the interface and stop the thread
     def stop
+      @cancel_thread = true
+      @thread_sleeper.cancel
       @interface.disconnect
 
       if @thread
         Logger.info "Stopping packet reading for #{@interface.name}"
-        @thread.kill
+        Cosmos.kill_thread(self, @thread) if @thread != Thread.current
```

Comment:
Ok I see how this same code can be called by the thread itself. Is there a reason the Logger.info line is inside the if. Don't we always want to report that we're stopping the packet reader?